File: PeakIndexInAMountainArray.java
class Solution {
    public int peakIndexInMountainArray(int[] A) {
        if (A.length <= 0) {
            return 0;
        }
        for (int i = 1; i < A.length - 1; i++) {
            if (A[i - 1] < A[i] && A[i] > A[i + 1]) {
                return i;
            }
        }
        return 0;
    }
}

File: PeakIndexInAMountainArray.go
/**
 * LeetCode 852 Peak Index in a Mountain Array
 */
func peakIndexInMountainArray(arr []int) int {
	if arr == nil || len(arr) == 0 {
		panic("param error")
	}
	return sort.Search(len(arr) - 1,func (i int) bool { return arr[i] > arr[i + 1]})
}
